eng/Signing.props

```diff
@@ -10,7 +10,8 @@
         to what is in when we run the Sign.proj. Corefx encodes more then just Debug/Release into the
         Configuration property so we simply need to glob everything under packages.
      -->
-    <ItemsToSign Include="$(ArtifactsDir)packages\**\*.nupkg" Exclude="$(ArtifactsDir)packages\**\*symbols.nupkg" />
+    <ItemsToSign Include="$(ArtifactsDir)packages\**\*.nupkg"
+                 Exclude="$(ArtifactsDir)packages\**\*symbols.nupkg;$(ArtifactsDir)packages\**\*.Private.*.nupkg" />
```

Comment:
@safern and I were chatting this morning that the "right way" to do this might be to partition the packages between Shipping/Not-shipping. *.Private.* is a heuristic that might be cheaper and good enough.  I know core-setup signs most of the content from the private packages, but I'm not sure if it does all.

@dagood do you know what core-setup signs?  Will it sign our reference assemblies and native binaries?

Reply:
Hmm, maybe not: https://github.com/dotnet/core-setup/blob/f9965e5a7eeb851ffd864f13915bea6c31a6fa72/signing/sign.proj#L29-L52

I think this would result in unsigned reference assemblies and native binaries (eg: clrcompression.dll)

Reply:
Yeah you're right, it seems like core-setup doesn't use arcade signing task which cracks open the nuget packages and signs its contents, it calls microbuild plugin task directly to sign packages, which just signs the nupkg's: https://github.com/dotnet/core-setup/blob/master/publish/publish.proj#L46

Reply:
Here's around where we sign Win x64 binaries in Core-Setup: 
https://dev.azure.com/dnceng/internal/_build/results?buildId=123276&view=logs&jobId=cb452c53-f418-5db4-86fa-6100fffaa4b0&taskId=ba7a9c1f-7cd9-5c4f-7aa1-2e75a24ba328&lineStart=97&lineEnd=97&colStart=9&colEnd=10

Looks like it signs the bits that Core-Setup actually builds plus the crossgen'd runtimes, then in the Finalize_Publish job we sign the nupkgs from all the platforms.

> Will it sign our reference assemblies and native binaries?

I agree it doesn't seem like it will, but I also don't think it makes sense for Core-Setup to be responsible for this--I've heard that as a general rule, signing should be done at the first possible point. (Better provenance with fewer systems involved.) Even delaying until the publish step is iffy (I think), but it's necessary for the stuff we build on non-Windows agents. 

I haven't been keeping up with the signing work that's happened in/since Arcade, so I might have outdated info on best practice.

(FYI, we don't have signing validation in Core-Setup at the moment, tracked by https://github.com/dotnet/core-setup/issues/5254. Just a heads up if someone wasn't aware.)

Reply:
Reading up a bit for context, what I'd suggest to avoid the perf issue and be "correct" by my understanding would be to continue to sign Windows bits on the Windows jobs and start *non-recursively* signing the nupkgs in the publish leg.

Reply:
after excluding *.Private.* signing step took around 8min which doesn't seem bad: https://dnceng.visualstudio.com/internal/_build/results?buildId=123609

I'm not much familiar with our infra but to me splitting signing in the same build seems confusing - when someone asks you where do you sign bits you know have 2 pieces to point to (possibly more in the future)

Reply:
I discussed with @safern and for now we'll just sign everything.  In the future we should change core-setup to sign all files that go into the shared frameworks, then we can avoid signing the transport packages and treat them as "non-shipping"

Reply:
Okay.  I took a look at the build times and it doesnt' look like a big increase overall, largely because the testing is done in parallel.

Reply:
Never mind my objections here, I was focusing too hard on some old thing I read. This looks fine to me.